### PR TITLE
use qutheory/console

### PR DESCRIPTION
**Pull request by tannernelson**
Add the qutheory/console protocol and terminal class.

### CODE REVIEW

File: Sources/Vapor/Application/Application.swift [hunk 1]
```diff
@@ -141,15 +141,25 @@ public class Application {
...
         self.arguments = arguments
 
-        let console = consoleProvided ?? Terminal()
+        let console = consoleProvided ?? Terminal(arguments: arguments)
         self.console = console
 
         let log = ConsoleLogger(console: console)
         self.log = log
 
+        func fileWorkDirectory() -> String? {
```
Comment (LoganWright) [inline Sources/Vapor/Application/Application.swift]: this seems like it should maybe be a global or static function, thoughts?

File: Sources/Vapor/Application/Application.swift [hunk 2]
```diff
@@ -250,78 +268,54 @@ extension Application {
...
-
-                if arguments.count < requiredArguments.count {
-                    let signature = command.dynamicType.signature()
-                    console.output(signature)
-                    throw ExecutionError.insufficientArguments
-                }
-
-                return command
-            }
+        if !args.flag("help") && args.values.count == 0 {
```
Comment (LoganWright) [inline Sources/Vapor/Application/Application.swift]: do we need to check for the flag? is it possible that count is 0, but a flag still exists?

Comment (tannernelson) [inline Sources/Vapor/Application/Application.swift]: Yeah, it's counting values which counts things that don't start with `--`

File: Sources/Vapor/Application/Application.swift (hunk 1 above)
Comment (tannernelson) [inline Sources/Vapor/Application/Application.swift]: Yeah, just where to put it is the question.

**Pull request merged**